kivy/uix/actionbar.py

```diff
@@ -486,24 +486,39 @@ def on_use_separator(self, instance, value):
             group.use_separator = value
         self.overflow_group.use_separator = value
 
+    def remove_widget(self, widget):
+        super(ActionView, self).remove_widget(widget)
+        if isinstance(widget, ActionOverflow):
+            for item in widget.list_action_item:
+                self._list_action_items.remove(item)
+
+        if widget in self._list_action_items:
+            self._list_action_items.remove(widget)
+
     def _clear_all(self):
+        lst = []
+        for item in self._list_action_items:
+            lst.append(item)
         self.clear_widgets()
         for group in self._list_action_group:
             group.clear_widgets()
 
         self.overflow_group.clear_widgets()
         self.overflow_group.list_action_item = []
+        self._list_action_items = lst
```

Comment:
I'm confused. What does it actually do? In the beginning of the method you append all items in `self._list_action_items` to that list, then clear all widgets in groups from the list and then return the same list? Isn't the list unchanged during the whole method?